### PR TITLE
Perfomance: Add prealloc linter and fix related allocation issues

### PR DESCRIPTION
**Pull request by peakle**
**What is this feature?**

Added `prealloc` linter for find preallocation issues on ci. Fix related allocation issues for slice/maps. This change allows us to reduce number of comments to PRs and write better codebase. Preallocations - decrease number of calls to the go allocator / OS api and thereby reduce the program execution time.

**Why do we need this feature?**

It will reduce number of comments to PRs and increase perfomance.

**Who is this feature for?**

It's perfomance improvement.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by peakle** [comment]
i can slice changes in few parts, if it difficult to read

**Comment by peakle** [comment]
i see that TestAlertManagers_buildRedactedAMs test failed because it have check for nil, should i revert code changes or update test?

**Comment by joanlopez** [comment]
Hey @peakle,

Any plan on continuing with this task? 
Could we consider it finished with the few smaller PRs already merged?

Thanks!

**Comment by peakle** [comment]
Hello @joanlopez, main purpose of pr was to add prealloc linter to grafana, i will try to merge actual master and check if there are any significant updates.

**Comment by joanlopez** [comment]
> Hello @joanlopez, main purpose of pr was to add prealloc linter to grafana, i will try to merge actual master and check if there are any significant updates.

Sounds good, thanks!

**Comment by chri2547** [comment]
No docs impact, removed myself as reviewer

File: .golangci.toml [hunk 1]
```diff
@@ -48,7 +48,6 @@ enable = [
   "exportloopref",
   "staticcheck",
   "stylecheck",
-  "typecheck",
```
Comment (peakle) [inline .golangci.toml:51]: it duplicates 56 line

**Comment by peakle** [comment]
Hey @joanlopez, merged all code changes, now only comments and config changes remain

**Comment by joanlopez** [comment]
> Hello @joanlopez, merged all code changes, now only comments and config changes remain

Hey @peakle, thanks for the huge effort, 
I think it's very positive to have migrated all these slices/maps initializations to preallocated memory!

That being said, I'm not sure if we do actually want to set up that linter and therefore require a `//nolint` comment for every single case where that's not possible (not easy to guess the amount of memory to allocate).

Let me ping @grafana/backend-platform , cause I think that question is a better fit for them.

Thanks!

**Pull request closed without merging**